### PR TITLE
Fix for issue #3802 

### PR DESCRIPTION
**Pull request by danieljames-dj**
I've made the change that jonathan told. I couldn't test it again because I couldn't run the php part in my system. I hope it won't have any syntax errors this time.

### CODE REVIEW

File: WcaOnRails/app/controllers/competitions_controller.rb [hunk 1]
```diff
@@ -266,6 +266,9 @@ def post_results
       unless comp.results
         return render html: "<div class='container'><div class='alert alert-warning'>No results</div></div>".html_safe
       end
+      unless comp.results_posted.nil?
```
Comment (jfly) [inline WcaOnRails/app/controllers/competitions_controller.rb]: I think this is causing test failures, see https://travis-ci.org/thewca/worldcubeassociation.org/builds/519622357#L6266

File: WcaOnRails/app/controllers/competitions_controller.rb [hunk 2]
```diff
@@ -342,9 +345,11 @@ def post_results
           body += "#{record_strs.join(", ")}.  \n" # Trailing spaces for markdown give us a <br>
         end
       end
-      comp.update!(results_posted_at: Time.now)
-      comp.competitor_users.each { |user| user.notify_of_results_posted(comp) }
-      comp.registrations.accepted.each { |registration| registration.user.notify_of_id_claim_possibility(comp) }
+      unless comp.results_posted?
```
Comment (jfly) [inline WcaOnRails/app/controllers/competitions_controller.rb]: By the time we get here, won't the results always be *not* posted (you check earlier if they are posted and bailed out)

**Comment by danieljames-dj** [comment]
Hi Jeremy, I'm sorry that I accidentally checked out new branch from wrong branch instead of master. Hence, one extra commit too came in the PR. I've submitted the new PR with just one commit.

**Comment by jonatanklosko** [comment]
Superseded by #3802.

**Pull request closed without merging**